sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderThread.kt

```diff
@@ -291,96 +317,124 @@ internal class MapboxRenderThread : Choreographer.FrameCallback {
     }
   }
 
+  @WorkerThread
+  internal fun processAndroidSurface(surface: Surface, width: Int, height: Int) {
+    if (this.surface != surface) {
+      if (this.surface != null) {
+        releaseEgl()
+        this.surface?.release()
+      }
+      this.surface = surface
+    }
+    this.width = width
+    this.height = height
+    // we clean only Mapbox events to avoid outdated runnables associated with previous EGL context
+    renderEventQueueLock.withLock {
+      // using iterator to avoid concurrent modification exception
+      val iterator = renderEventQueue.iterator()
+      while (iterator.hasNext()) {
+        val next = iterator.next()
+        if (next.eventType == EventType.SDK) {
+          iterator.remove()
+        }
+      }
+    }
+    // we do not want to clear render events scheduled by user
+    renderHandlerThread.clearMessageQueue(clearAll = false)
+    prepareRenderFrame(creatingSurface = true)
+  }
+
   @UiThread
   fun onSurfaceCreated(surface: Surface, width: Int, height: Int) {
     lock.withLock {
       renderHandlerThread.post {
-        if (this.surface != surface) {
-          releaseEgl()
-          this.surface?.release()
-          this.surface = surface
-        }
-        this.width = width
-        this.height = height
-        shouldExit = false
-        eventQueue.clear()
-        renderEventQueue.clear()
-        snapshotQueue.clear()
-        renderHandlerThread.clearMessageQueue()
-        prepareRenderFrame(creatingSurface = true)
+        processAndroidSurface(surface, width, height)
       }
       createCondition.await()
     }
   }
 
   @AnyThread
   fun setMaximumFps(fps: Int) {
-    renderTimeNs.set(ONE_SECOND_NS / fps)
+    renderTimeNs = ONE_SECOND_NS / fps
```

Comment:
Maybe update it similarly to size - posting runnable to the render thread? No need for volatile then

Reply:
imo code is easier with volatile, in general does not matter that much but slightly easier tests now